aten/src/THC/THCAtomics.cuh

```diff
@@ -12,36 +12,46 @@ struct AtomicAddIntegerImpl;
 template<typename T>
 struct AtomicAddIntegerImpl<T, 1> {
   inline __device__ void operator()(T *address, T val) {
-    uint32_t * address_as_ui =
-        (uint32_t *) (address - ((size_t)address & 3));
+    size_t offset = (size_t)address & 3;
```

Comment:
There's some rampant casting going on in this function. I know that we don't follow any style guides whatsoever, but in general, c-style casts are not very "c++": https://google.github.io/styleguide/cppguide.html#Casting . What are your thoughts on this subject?

Reply:
I think I prefer the explicit style in those examples. I'd like to see us move towards having more formal style guides and I think there are good reasons for C++ style to differ from C. 

One concern though is I'm fixing a small bug in two of these addIntegerImpl's, but there are a number of these which all look very similar. Things that make this particular impl easier to read could also make it inconsistent with the others. If I go too far in changing the style I might prefer to do that in a separate PR and do all impls.